### ISSUE TITLE
Adding custom Javascript to Just Read pages

### ISSUE THREAD

**Issue by StevenTammen** [ZachSaucier/Just-Read#216]
I came across your app while researching replacements for Evernote clearly that allowed for custom CSS. I'm pleased with how things work, but realized a few articles in that there are several Javascript libraries on some of the pages I commonly read that don't make it into the iframe.

For example, some websites make use of a [Bible-verse tagger](https://faithlife.com/products/reftagger) that shows tooltip windows when hovering over verse references. I haven't looked at the exact implementation, but it appears to do a couple things:

1. Use regular expressions to traverse the DOM on page load, replacing matching text with links to a particular site's API (also giving these links a special class).
2. Register event handlers for this class of links on link hover. The handlers seem to use appendChild() to add `<div>`'s with custom classes to the bottom of `<body>`. After the cursor leaves the tooltip window, these added `<div>`'s are then deleted.

(1) Seems to work fine with Just Read (although I haven't tried it with Just Read automatically activating on pages: there might be a race condition here). (2), however, does not work at all.

A very similar case happens with footnotes that use tooltips. For example, have a look at the page for this [jQuery Footnote Plugin](http://bigfootjs.com/). Without any Javascript in the iframe, the footnotes do not do anything.

----------------------------

I am relatively new to web programming (and iframes in particular), and it is unclear to me if it would be possible to have behavior like this work for pages transformed with Just Read. I understand that this application is focused on web pages that take an article form, and hence doesn't concern itself with source page's Javascript. However, I can see situations in which Javascript is actually important for reading the page properly (as above), and it seems to me like it would be very useful to be able to include it in the iframe in some situations. (Either injecting custom Javascript into all Just Read pages, or injecting specific Javascript on specific pages that match regular expressions).

What do you think? I have some more detailed ideas, but will hold off on further elaboration for the moment.

(P.S. -- It's cool to see something this developed come out of UGAHacks. I went last year -- I'm a CS student at UGA -- and we didn't make anything nearly so impressive.)

**Comment by ZachSaucier** [comment]
Hey Steven and thanks for posting,

While I see the value in including tooltips like these on hover, including tooltips generically in the JR version would be a very difficult task because it's very difficult to tell which scripts are relevant and include only those specifically. There are many types of these hover-tooltips, including custom versions that individual sites will have. As such, I don't think it's within the current scope of Just Read to implement functionality like this. 

With that being said, you could write a userscript to add this functionality to JR for a specific domain. In essence, you'd manually look for the relevant JS file (and hope that it is in its own JS file, lol) or copy only the relevant part of a larger JS file, then write a userscript to detect when JR is started on that domain and inject the relevant part of the original page's JS to recreate the tooltip functionality. 

As you can tell from this process, it would be difficult to do generically 😛 

___

TL;DR: JR is meant to take an already-rendered article and put it into a JR version. So anything dynamic like these tooltips is really out of scope currently. As such, I'm going to close this request.
